src/alchemlyb/visualisation/ti_dhdl.py

```diff
@@ -38,17 +42,26 @@ def plot_ti_dhdl(dhdl_data, labels=None, colors=None, units='kcal/mol', ax=None)
     matplotlib.axes.Axes
         An axes with the TI dhdl drawn.
 
-    Notes
-    -----
+    Note
+    ----
     The code is taken and modified from
     : `Alchemical Analysis <https://github.com/MobleyLab/alchemical-analysis>`_
 
+    The units variable is for labelling only. Changing it doesn't change the
+    unit of the underlying variable, which is in the unit of kT. The
+    scaling_factor is used to change the number to the desired unit.
+
     '''
+    # Fix unit
+
+
     # Make it into a list
-    try:
-        len(dhdl_data)
-    except TypeError:
-        dhdl_data = [dhdl_data, ]
+    if not isinstance(dhdl_data, list):
+        dhdl_list = dhdl_data.separate_dhdl()
+    else:
+        dhdl_list = []
+        for dhdl in dhdl_data:
+            dhdl_list.extend(dhdl.separate_dhdl())
```

Comment:
Using the new function improves the code, so that's good.

Perhaps add a comment what is done here and why, i.e., explain the code logic.